### PR TITLE
futures: replace BatchStream with BatchResultStream

### PR DESCRIPTION
**Pull request by huachaohuang**
Check a use case of this stream in #252 .

### CODE REVIEW

**Review by zojw — DISMISSED** [review]

File: src/storage/src/local/mem/storage.rs [hunk 1]
```diff
@@ -33,15 +33,14 @@ use crate::{async_trait, Error, Result};
...
 }
 
 impl Default for Storage {
     fn default() -> Self {
         Self {
-            buckets: Arc::new(Mutex::new(HashMap::new())),
+            buckets: Mutex::new(HashMap::new()),
         }
     }
 }
```
Comment (tisonkun) [inline src/storage/src/local/mem/storage.rs:46]: unrelated changes?

Comment (huachaohuang) [reply src/storage/src/local/mem/storage.rs:46]: Yeah, I can open another PR for this if necessary.

Comment (tisonkun) [reply src/storage/src/local/mem/storage.rs:46]: @huachaohuang It's OK or you can separate in a new commit and use "rebase and commit". I don't insist a new PR but distinguish in commits.

**Review by w41ter-l — APPROVED** [review]
